docs/content/guides/developer/app-examples/reviews-rating.mdx

```diff
@@ -0,0 +1,212 @@
+---
+title: Reviews Rating
+---
+
+The following documentation goes through an example implementation of a review rating platform for the food service industry on Sui.
+Unlike traditional review rating platforms, that often do not disclose the algorithm used to rate reviews, the on-chain platform uses an algorithm that is published on-chain for everyone to see and verify.
+Furthermore, all submitted reviews are scored and ordered on-chain which is feasible due to the low gas cost of computation on Sui.
+
+## Personas
+
+There are four actors in the typical workflow of the reviews rating example.  
+
+```mermaid
+sequenceDiagram
+  Service ->> Dashboard: Add service to dashboard.
+  Service ->> User: Send proof of experience.
+  User ->> Service: Send review
+  Service ->> User: Send reward
+  Service ->> Moderator: Add as a service moderator
+  Moderator ->> Service: Remove abused review
+```
+
+### Service owners
+
+Service owners are entities like restaurants that list their services on the platform. Service owners allocate a specific amount of SUI as a reward pool. Assets from the pool are used to provide rewards for high-rated reviews. A Proof of Experience (PoE) NFT confirms the reviewer used the service, which they burn to provide a verified review. Service owners can provide their customers with unique identifiers, perhaps using QR codes, to identify individual reviewers.
+
+### Reviewers
+
+Reviewers are consumers of services that use the review system. Reviewers provide feedback in the form of comments that detail specific aspects of the service as well as a star rating to inform others. The reviews are rated, with the most effective reviews getting the highest rating. Service owners award the 10 highest rated reviews for their service. How often the rewards are distributed is up to the service owner; it can be on a regular or rolling basis.
+
+### Review readers
+
+Review readers access reviews to make informed decisions on selecting services. Readers rate reviews by casting up or down votes. The review readers' ratings are factored into the algorithm that rates the reviews, with the authors of the highest-rated reviews getting awarded. 
+
+### Moderators
+
+Moderators monitor content of the reviews and can delete any reviews that contain inappropriate content.
+Moderators also keep an eye on the number of reviews registered for a service to ensure that they do not exceed the maximum number allowed.
+When the maximum number of reviews is reached, moderators can prune old reviews to make room for a new review.
+
```

Comment:
The incentives piece for reviewers and moderators seems missing to me. It's clear why service owners and reviewers want to participate in this system, but why do readers and moderators want to spend time or gas here?

Reply:
I guess the usual playbook would be to distribute rewards to these groups as well, e.g.

- Review readers get a proportion of the review reward if they voted for it (you may need to make votes scarce for this to have an effect, or you may not given that people need to take the time and pay gas to place a vote).
- Service providers all pay into a pool that goes to Moderators, on a rolling basis, people can stake moderators to influence what proportion of the reward each moderator gets, up to a limit (similar to how validators are staked on-chain), and moderator decisions are decided by quorum (of stake weight).

These things can be complex to implement though, so I think it would be fine to address the question of incentives and say that you *could* do these things, but they are not implemented as part of this guide.

Reply:
Added incentives for review readers and moderators as suggested.